traversal/structure/Structure.java

```diff
@@ -82,11 +90,21 @@ public void equalEdge(StructureVertex<?> from, StructureVertex<?> to) {
         recordEdge(edge);
     }
 
-    public void predicateEdge(StructureVertex.Thing from, StructureVertex.Thing to, Predicate.Variable predicate) {
+    public void predicateEdge(StructureVertex<?> from, StructureVertex<?> to, Predicate.Variable predicate) {
         StructureEdge.Predicate edge = new StructureEdge.Predicate(from, to, predicate);
         recordEdge(edge);
     }
 
+    public void dependencyEdge(StructureVertex.Value operand, StructureVertex.Value result) {
```

Comment:
Yeah, I thought I'd need dependencies elsewhere (in predicates) as well so I figured it could be a single dependency edge at this level/the planner level and then the actual edge in the procedure. I'll change these to argument as well.